File: cpp/intervals/merge_intervals.cpp
/*
Problem: Merge Intervals
Approach:
1. Sort intervals by start then merge O(n log n)
*/
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        // TODO: implement
        return {}; // placeholder
    }
};

int main() {
    vector<vector<int>> intervals = {{1,3},{2,6},{8,10},{15,18}};
    Solution s; auto merged = s.merge(intervals);
    for (auto &v : merged) { cout << '[' << v[0] << ',' << v[1] << "] "; }
    cout << endl;
}

File: python/intervals/merge_intervals.py
"""
Problem: Merge Intervals
Given an array of intervals where intervals[i] = [start, end], merge all overlapping intervals.

Approach:
1. Sort by start then iterate and merge O(n log n)
"""
from typing import List

class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        # TODO: implement
        raise NotImplementedError

if __name__ == "__main__":
    s = Solution()
    print(s.merge([[1,3],[2,6],[8,10],[15,18]]))  # expect [[1,6],[8,10],[15,18]]
